src/main/java/com/squareup/auto/value/redacted/AutoValueRedactedExtension.java

```diff
@@ -0,0 +1,129 @@
+package com.squareup.auto.value.redacted;
+
+import com.google.auto.service.AutoService;
+import com.google.auto.value.AutoValueExtension;
+import com.google.common.collect.Lists;
+import com.squareup.javapoet.ArrayTypeName;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.JavaFile;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterSpec;
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+import com.squareup.javapoet.TypeVariableName;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.Name;
+
+@AutoService(AutoValueExtension.class)
+public class AutoValueRedactedExtension implements AutoValueExtension {
+
+  @Override
+  public boolean applicable(Context context) {
+    Map<String, ExecutableElement> properties = context.properties();
+    for (ExecutableElement element : properties.values()) {
+      if (isRedacted(element)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  @Override
+  public boolean mustBeAtEnd(Context context) {
+    return false;
+  }
+
+  @Override
+  public String generateClass(Context context, String className, String classToExtend,
+      boolean isFinal) {
+    Name superName = context.autoValueClass().getSimpleName();
+    Map<String, ExecutableElement> properties = context.properties();
+
+    TypeSpec subclass = TypeSpec.classBuilder(className) //
+        .addModifiers(isFinal ? Modifier.FINAL : Modifier.ABSTRACT) //
+        .superclass(TypeVariableName.get(classToExtend)) //
```

Comment:
What's the `TypeVariableName` here? I think you want `ClassName.bestGuess` instead.